src/ui/movable_rect.cpp

```diff
@@ -0,0 +1,44 @@
+#include <ui/movable_rect.hpp>
+
+namespace nebula {
+  Movable_Rect::Movable_Rect(void (*render_function)(Rect_f32))
+  {
+    position = {-0.1f, 0.1f, 0.1f, -0.1f};
+    this->render_function = render_function;
+  }
+
+  void Movable_Rect::Movable_Rect::render() const
+  {
+    render_function(position);
+  }
+
+  bool Movable_Rect::is_under_mouse(const Vec2 mouse_location,
+                                    const f32 window_width,
+                                    const f32 window_height) const
+  {
+    if(mouse_location.x >=
```

Comment:
1. break it up into multiple bool vars with meaningful names.
2. if(x) { return true } else { return false } => return x [significantly better and shorter]

Reply:
Used in the pixel struct method